src/async_fuse/memfs/cache/backend.rs

```diff
@@ -0,0 +1,493 @@
+        let path = Self::get_block_path(ino, block_id);
+
+        let block_start = block.start();
+        let block_end = block.end();
+
+        if block_start == 0 && block_end == self.block_size {
+            // To store a whole block
+            let mut writer = self.operator.writer(&path).await.unwrap_or_else(|e| {
+                panic!(
+                    "Failed to get a writer to backend where ino={ino} and block={block_id}: {e}"
+                )
+            });
+            writer.write_all(block.as_slice()).await.unwrap_or_else(|e| panic!("Failed to store a block to backend where ino={ino} and block={block_id}: {e}"));
+            writer.close().await.unwrap_or_else(|e| {
+                panic!("Failed to close the writer where ion={ino} and block={block_id}: {e}")
+            });
+            return;
+        }
+
+        let mut dest = self.operator.read(&path).await.unwrap_or_else(|e| {
+            if e.kind() == ErrorKind::NotFound {
+                // Create an empty block for overwriting is ok.
+                vec![]
+            } else {
+                panic!("Failed to load a block to backend for overwrite where ino={ino} and block={block_id}: {e}");
+            }
+        });
+
+        // Ensure that the vector is long enough to be overwritten
+        if dest.len() < block_end {
+            dest.resize(block_end, 0);
+        }
+
+        // merge two blocks
+        dest.get_mut(block_start..block_end)
+            .unwrap_or_else(|| unreachable!("The vector is ensured to be long enough."))
+            .copy_from_slice(block.as_slice());
+        self.operator.write(&path, dest).await.unwrap_or_else(|e| {
+            panic!("Failed to store a block to backend where ino={ino} and block={block_id}: {e}")
+        });
+    }
+
+    async fn remove(&self, ino: INum) {
+        self.operator
+            .remove_all(&Self::get_file_path(ino))
+            .await
+            .unwrap_or_else(|e| panic!("Failed to remove a file of ino={ino}: {e}"));
+    }
+
+    async fn invalidate(&self, _: INum) {
+        // This storage has no cache, therefore, its contents cannot be
+        // invalidated.
+    }
+
+    async fn flush(&self, _: INum) {
+        // This storage has no cache and backend, therefore, there is no need to
+        // flush its data.
+    }
+
+    async fn flush_all(&self) {
+        // This storage has no cache and backend, therefore, there is no need to
+        // flush its data.
+    }
+
+    async fn truncate(&self, ino: INum, from_block: usize, to_block: usize, fill_start: usize) {
+        let paths =
+            stream::iter(to_block..from_block).map(|block_id| Self::get_block_path(ino, block_id));
+
+        let file_path = Self::get_file_path(ino);
+
+        let file_exists = self
+            .operator
+            .is_exist(&file_path)
+            .await
+            .unwrap_or_else(|e| panic!("Failed to check if file ino={ino} exists: {e}"));
```

Comment:
Can we treat this error as "not exist" instead of panic?

Reply:
Ok, I will fix it later.